src/metrics-modeling/metricbot/api.py

```
@@ -0,0 +1,42 @@
+def post_predictions(config: Config, dtos: list[dict], timeout_seconds: int = 60) -> str:
+    url = f"{config.api_base_url}/admin/ai-predictions/{config.metricbot_user_id}"
+
+    request = urllib.request.Request(
+        url,
+        data=json.dumps(dtos).encode("utf-8"),
+        headers={
+            "Content-Type": "application/json",
+            "X-Admin-Token": config.admin_token,
+        },
+        method="POST",
+    )
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)